src/server_manager/web_app/app.ts

```diff
@@ -609,21 +608,29 @@ export class App {
 
   private showTransferStats(runningServer: server.Server, serverView: Polymer) {
     const refreshTransferStats = () => {
-      runningServer.getDataUsage().then((stats) => {
-        // Calculate total bytes transferred.
-        let totalBytes = 0;
-        // tslint:disable-next-line:forin
-        for (const accessKeyId in stats.bytesTransferredByUserId) {
-          totalBytes += stats.bytesTransferredByUserId[accessKeyId];
-        }
-        serverView.setServerTransferredData(totalBytes);
-        // tslint:disable-next-line:forin
-        for (const accessKeyId in stats.bytesTransferredByUserId) {
-          const transferredBytes = stats.bytesTransferredByUserId[accessKeyId];
-          const relativeTraffic = totalBytes ? 100 * transferredBytes / totalBytes : 0;
-          serverView.updateAccessKeyRow(accessKeyId, {transferredBytes, relativeTraffic});
-        }
-      });
+      runningServer.getDataUsage().then(
+          (stats) => {
+            // Calculate total bytes transferred.
+            let totalBytes = 0;
+            // tslint:disable-next-line:forin
+            for (const accessKeyId in stats.bytesTransferredByUserId) {
+              totalBytes += stats.bytesTransferredByUserId[accessKeyId];
+            }
+            serverView.setServerTransferredData(totalBytes);
+            // tslint:disable-next-line:forin
+            for (const accessKeyId in stats.bytesTransferredByUserId) {
+              const transferredBytes = stats.bytesTransferredByUserId[accessKeyId];
+              const relativeTraffic = totalBytes ? 100 * transferredBytes / totalBytes : 0;
+              serverView.updateAccessKeyRow(accessKeyId, {transferredBytes, relativeTraffic});
+            }
+          },
+          (e: errors.ServerApiError) => {
+            // Since failure is invisible to users, allow non-network errors percolate to the top
+            // so that a Sentry report will be generated.
+            if (e.response) {
```

Comment:
Do we want a whitelist or blacklist approach?
I remember you said you wanted to suppress network errors, but you implemented propagation of api errors, which is not quite the same. Should we have a `NetworkError` instead?

I guess there's a question about what is special about those errors that make it ok to ignore them. Is it because they are transient errors? If so, should we have a `TransientError` instead?

Those seem minor, but it's important to align the code with what we care about in the product.